shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/core/NetworkClient.kt

```diff
@@ -0,0 +1,58 @@
+package co.nimblehq.blisskmmic.data.network.core
+
+import co.nimblehq.jsonapi.json.JsonApi
+import io.ktor.client.HttpClient
+import io.ktor.client.call.*
+import io.ktor.client.engine.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.plugins.logging.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.serialization.json.Json
+
+class NetworkClient {
+
+    val client: HttpClient
+
+    val json = Json {
+        prettyPrint = true
+        isLenient = true
+        ignoreUnknownKeys = true
+    }
+
+    constructor(engine: HttpClientEngine? = null) {
+        if (engine == null) {
+            client = HttpClient() {
+                install(Logging)
+                install(ContentNegotiation) {
+                    json(Json {
+                        prettyPrint = true
+                        isLenient = true
+                        ignoreUnknownKeys = true
+                    })
+                }
+            }
+        } else {
+            client = HttpClient(engine) {
+                install(Logging)
+                install(ContentNegotiation) {
+                    json(json)
+                }
+            }
+        }
+    }
```

Comment:
I don't really get the idea, why do we need to init `HttpClient` with null `engine` case? They should go along together from my understanding 🤔

Reply:
This is for DI the engine, the issue is that `HttpClient()` will automatically create engine that is specific for the platform it is running, if we pass engine to `HttpClient(engine)` we will need to declare engine for each platforms (Android, iOS) and not using the default engine. 
Creating the constructor like this save the code in declaring engine for each of the platforms.

Reply:
@blyscuit Great, I will try out this 👍